vaadin-details-flow-parent/vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java

```diff
@@ -321,8 +322,17 @@ public boolean isOpened() {
      */
     public Registration addOpenedChangeListener(
             ComponentEventListener<OpenedChangeEvent> listener) {
-        return getElement().addPropertyChangeListener("opened",
-                event -> listener.onComponentEvent(
-                        new OpenedChangeEvent(this, event.isUserOriginated())));
+        if (openedPropertyChangeEventHandler == null) {
+            openedPropertyChangeEventHandler = new PropertyChangeEventHandler<>(
+                    "opened", this, OpenedChangeEvent.class) {
+                @Override
+                protected void fireComponentEvent(
+                        PropertyChangeEvent propertyChangeEvent) {
+                    Details.this.fireEvent(new OpenedChangeEvent(Details.this,
+                            propertyChangeEvent.isUserOriginated()));
+                }
+            };
+        }
+        return openedPropertyChangeEventHandler.addListener(listener);
     }
```

Comment:
Looking at how `ComponentEvent.unregisterListener` works, it depends on `ComponentEventBus` to set the unregister listener:
https://github.com/vaadin/flow/blob/3aaaeea838d6b120dd2e717bb1891479c0a2980e/flow-server/src/main/java/com/vaadin/flow/component/ComponentEventBus.java#L227-L235

So rather than implementing our custom solution here, maybe we should reuse `ComponentEventBus`. I was able to get the tests to pass by making two changes. First, define the `opened` property change listener once in the constructor, and make it fire the `OpenedChangeEvent`:
```java
getElement().addPropertyChangeListener("opened",
        event -> fireEvent(new OpenedChangeEvent(this, event.isUserOriginated())));
```
Then `addOpenedChangeListener` just becomes:
```java
public Registration addOpenedChangeListener(
        ComponentEventListener<OpenedChangeEvent> listener) {
    return addListener(OpenedChangeEvent.class, listener);
}
```
Both `fireEvent` and `addListener` delegate to the component event bus, which makes `unregisterListener` work automatically.